tools/pyboard.py

```diff
@@ -258,7 +316,12 @@ def __init__(
         self.in_raw_repl = False
         self.use_raw_paste = True
         if device.startswith("exec:"):
-            self.serial = ProcessToSerial(device[len("exec:") :])
+            # Class selector based on select module platform dependancy (available on Unix, but not on Windows)
+            import select
+            if hasattr(select, "poll"):
+                self.serial = ProcessToSerialPosix(device[len("exec:") :])
+            else:
+                self.serial = ProcessToSerialThreading(device[len("exec:") :])
```

Comment:
The `ProcessToSerial` class is really part of the public API of this file, so I suggest to put this code in the global scope, so that `ProcessToSerial` is defined in the global scope.  Then the code here doesn't need to change at all.

Reply:
Excuse me, I don't have much experience with Python. If I keep the ProcessToSerial class, how and where I could declare ProcessToSerialPosix and ProcessToSerialThreading ? I appreciate very much your help and indications.
Thank you in advance.
Best regards,
Victor

Reply:
```python
class ProcessToSerialPosix:
    ...

class ProcessToSerialThreading:
    ...

if hasattr(select, "poll"):
    ProcessToSerial = ProcessToSerialPosix
else:
    ProcessToSerial = ProcessToSerialThreading
```

Reply:
Thank you very much for your time and help ! I didn't know you can do that in Python. I made a commit with these changes.